app/models/mhv_account.rb

```diff
@@ -109,29 +112,41 @@ def create_mhv_account!
     if may_register?
       client_response = mhv_ac_client.post_register(params_for_registration)
       if client_response[:api_completion_status] == 'Successful'
+        StatsD.increment("#{STATSD_ACCOUNT_CREATION_KEY}.success")
+
         user.va_profile.mhv_ids = [client_response[:correlation_id].to_s]
         user.instance_variable_get(:@mvi).save
         self.registered_at = Time.current
         register!
       end
       # TODO: be prepared to handle or raise various exceptions
     end
+  ensure
+    StatsD.increment("#{STATSD_ACCOUNT_CREATION_KEY}.total") if may_register? || registered?
   end
 
   def upgrade_mhv_account!
-    if may_upgrade?
+    initially_premium = !may_upgrade?
+
+    unless initially_premium
```

Comment:
prefer this to be `if may_upgrade?` since we call this method whether you're initially premium and also if you're registered but not yet premium.